pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -108,6 +109,7 @@ type HintProvider interface {
 // which can allocate resources, provide allocation hints.
 type ResourceAllocator interface {
 	HintProvider
+	AllocationInfoProvider
```

Comment:
Rather than creating  a new interface (ResourceAllocator) why don't we embed `AllocationInfoProvider` directly in HintProvider interface or add `GetExclusiveResources` to the HintProvider interface?

Reply:
This stems from my interpretation of a previous comment from @klueska . We can shoehorn the AllocationInfoProvider into the HintProvider, but feels like... forced. OTOH the two interfaces are related, so the option I like more atm is to create a new abstraction which includes both.

I'm open to suggestions though.

Reply:
Thanks for the explanation, I was mostly curious and wanted to understand the thought process here.